### ISSUE TITLE
Unable to Put a Wheel in requirements.in File

### ISSUE THREAD

**Issue by cancan101** [jazzband/pip-tools#415]
One of my requirements is not on pypi but does offer a URL to a wheel.

##### Steps to replicate

1.  Put wheel URL in `in` file and try to `pip-compile`.
2. Put wheel URL in `in` file with editable and try to `pip-compile`

##### Expected result
Compile to txt file.

##### Actual result
Without editable:
```
pip.exceptions.InstallationError: https://storage.googleapis.com/tensorflow/mac/gpu/tensorflow-0.11.0-py2-none-any.whl should either be a path to a local project or a VCS url beginning with svn+, git+, hg+, or bzr+
```
with editable:
```
Traceback (most recent call last):
  File "/Users/alex/.virtualenvs/test/bin/pip-compile", line 11, in <module>
    sys.exit(cli())
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/piptools/scripts/compile.py", line 180, in cli
    src_file, finder=repository.finder, session=repository.session, options=pip_options))
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/pip/req/req_file.py", line 93, in parse_requirements
    for req in req_iter:
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/pip/req/req_file.py", line 168, in process_line
    wheel_cache=wheel_cache
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/pip/req/req_install.py", line 144, in from_editable
    editable_req, default_vcs)
  File "/Users/alex/.virtualenvs/test/lib/python2.7/site-packages/pip/req/req_install.py", line 1183, in parse_editable
    editable_req
pip.exceptions.InstallationError: https://storage.googleapis.com/tensorflow/mac/gpu/tensorflow-0.11.0-py2-none-any.whl should either be a path to a local project or a VCS url beginning with svn+, git+, hg+, or bzr+
```

This may be related: https://github.com/nvie/pip-tools/pull/213

**Comment by davidovich** [comment]
This is a use case implemented in PR #213, but has not gained any traction. It is a bit outdated, but the actualisation work shouldn't be hard.

**Comment by atugushev** [comment]
It's implemented in #807:

```bash
$ pip-compile --version
pip-compile, version 4.1.0

$ echo https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl \
    | pip-compile - -qo-
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file=- -
#
https://files.pythonhosted.org/packages/fa/37/45185cb5abbc30d7257104c434fe0b07e5a195a6847506c074527aa599ec/Click-7.0-py2.py3-none-any.whl
```
